rmgpy/tools/thermo_group_validation.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+This script compares the group additivity to species in a thermo library. It is also used to compare
+differences in group additivity between different commits of RMG-database.
+"""
+from contextlib import contextmanager
+import os
+import csv
+import subprocess
+from numpy import std
+import argparse
+from rmgpy import settings
+from rmgpy.data.thermo import ThermoDatabase, ThermoLibrary
+from rmgpy.species import Species
+import matplotlib.pyplot as plt
+
+#initalize input variables
+global speciesThermo, inputDict
+speciesThermo = {}
+inputDict = {}
+
+
+@contextmanager
+def cd(newdir):
+    """
+    Context to change directory to newdir before reverting back to the original working directory
+    """
+    prevdir = os.getcwd()
+    os.chdir(os.path.expanduser(newdir))
+    try:
+        yield
+    finally:
+        os.chdir(prevdir)
+
+@contextmanager
+def previousCommit(commitStr, gitDirectory):
+    """
+    Context to go to a previous commit before reverting back to the the original commit
+    """
+    if commitStr == 'HEAD': yield
+    else:
+        try:
+            #Check that the commit str is not a file, so that you don't accidently check out unsaved changes
+            if os.path.exists(commitStr):
+                raise Exception("Argument commitStr must be a sha1, not a file")
+            else:
+                subprocess.call(['git', 'checkout', commitStr])
+                actualCommit = subprocess.check_output(['git', 'log','--format=%H%n%cd', '-1'], cwd=gitDirectory).splitlines()[0]
+                assert commitStr in actualCommit, "Checkout to another commit failed, please make sure your changes are stashed or committed."
```

Comment:
if you have some uncommitted changes on top of HEAD, maybe you should also give assertion warning?